src/client/camera.cpp

```diff
@@ -694,13 +695,31 @@ void Camera::drawNametags()
 			i = m_nametags.begin();
 			i != m_nametags.end(); ++i) {
 		Nametag *nametag = *i;
+		core::vector3df cam_pos = m_cameranode->getAbsolutePosition();
+		core::vector3df nametag_pos = nametag->parent_node->getAbsolutePosition();
+
+		if (!nametag->nametag_viewthrough_walls) {
+			if (cam_pos.getDistanceFrom(nametag_pos) >
+					stoi(g_settings->get("viewing_range")) &&
+					cam_pos.getDistanceFrom(nametag_pos) >
+					stoi(g_settings->get("player_transfer_distance")))
```

Comment:
Checking `player_transfer_distance` here is incorrect because that is the client setting and also unnecessary because the server already skips sending objects that are too far.

Reply:
objects including the player?

Reply:
Removing...

Reply:
Yes, that includes players.

Reply:
Sorry to unresolve, check is still present, best to resolve when code is updated.

Reply:
Done now